KnowledgeExtractor/WikipediaKnowledgeGraphExtractor.cs

```diff
@@ -96,34 +101,70 @@ public static KnowledgeGraph ExtractKnGraphFromUris(List<UriAndOriginalGraphType
         public static KnowledgeGraph GetKnowledgeGraphFromWikipedia()
         {
             KnowledgeGraph result = new KnowledgeGraph();
-
-            // get the kn graph from the uris
             result = ExtractKnGraphFromUris(WikipediaPagesToParse);
+            AddEdgesBetweenAlgorithmsAndDataStructures(ref result);
+            return result;
+        }
+
+        public static void AddEdgesBetweenAlgorithmsAndDataStructures(ref KnowledgeGraph graph)
+        {
+            List<WordCount> dataStructureWordsInGraph = GetDataStructureWordsInEachWikiPage();
+            Dictionary<string, KnGNode> dataStructureNodes = GetDataStructureNodesWithoutDuplicates(graph);
+            
+            foreach (var dataStructureWordsToAddInOneNode in dataStructureWordsInGraph)
+            {
+                AddDataStructureWordsToNode(graph: ref graph, wordsToAddToNode: dataStructureWordsToAddInOneNode, dataStructureNodes: dataStructureNodes);
```

Comment:
The named parameters don't really help readability here, since the parameter names match the values pretty exactly already. I'd remove the explicit parameter names